components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/ApplicationImportExportManager.java

```diff
@@ -188,6 +216,41 @@ private boolean isTierAvailable(Tier targetTier, API api) {
         }
     }
 
+    /**
+     * Check whether the object is a type of ApiProduct
+     *
+     * @param object        - {@link Object}
+     * @return true, if the object is an ApiProduct, otherwise false
+     */
+    private boolean isApiProduct(Object object) {
+        try {
+            //Cast object to ApiProduct
+            APIProduct apiProduct = (APIProduct) object;
+            return (apiProduct != null) ? true : false;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    /**
+     * Check whether a target Tier is available to subscribe
+     *
+     * @param targetTier Target Tier
+     * @param apiProduct - {@link APIProduct}
+     * @return true, if the target tier is available
+     */
+    private boolean isTierAvailableForProduct(Tier targetTier, APIProduct apiProduct) {
+        APIProductIdentifier apiProductId = apiProduct.getId();
+        Set<Tier> availableTiers = apiProduct.getAvailableTiers();
+        if (availableTiers.contains(targetTier)) {
+            return true;
+        } else {
+            log.error("Tier:" + targetTier.getName() + " is not available for API Product " + apiProductId.getName() + "-" +
+                    apiProductId.getVersion());
+            return false;
+        }
+    }
+
```

Comment:
I think we don't need another function like this. We can modify the existing isTierAvailable function as below. Just check the feasibility whether we can do like this.
```
private boolean isTierAvailable(Tier targetTier, ApiTypeWrapper apiTypeWrapper) {
	Set<Tier> availableTiers = null;
	if (!apiTypeWrapper.isAPIProduct()) {
		APIIdentifier apiId = apiTypeWrapper.getApi().getId();
		availableTiers = api.getAvailableTiers();
	} else {
		APIProductIdentifier apiProductId = apiTypeWrapper.getApiProduct().getId();
		availableTiers = apiProductId.getAvailableTiers();
	}
        if (availableTiers.contains(targetTier)) {
            return true;
        } else {
            log.error("Tier:" + targetTier.getName() + " is not available for API " + apiId.getApiName() + "-" +
                    apiId.getVersion());
            return false;
        }
}
```

Reply:
Fixed